fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -786,6 +810,50 @@ protected  <T> CompletableFuture<Void> iterateRangeOnly(@Nonnull FDBRecordStore
                 });
     }
 
+    private CompletableFuture<Void> validateTypeStamp(@Nonnull FDBRecordStore store) {
+        final long minimalInterval = policy.getCheckIndexingMethodFrequencyMilliseconds();
+        if (minimalInterval < 0 || isScrubber) {
+            return AsyncUtil.DONE;
+        }
+        if (minimalInterval > 0) {
+            final long now = System.currentTimeMillis();
+            if (now < lastTypeStampCheckMillis + minimalInterval) {
+                return AsyncUtil.DONE;
+            }
+            lastTypeStampCheckMillis = now;
+        }
```

Comment:
Since blocking is not expected to be immediate or atomic, it seems reasonable to let the caller choose the desire frequency - every transaction, clock, or never (zero, positive, or negative numbers respectively) according to his needs. 
Since we were struggling with indexing performance recently, it seems reasonable to set the default to a non-disruptive 1 minute wait between checks. `SynchronizedSession`, for comparison, has a default of 10 seconds before expiring. 
note that `SynchronizedSession` is only applied by the runner and only to the primary index. This check is performed for every target index.

Reply:
I see. I think that's fine, as long as this isn't critical for correctness, though the exact semantics were a bit unclear to me.